### PR TITLE
cein/feature/add-iam-permissions-boundary-input

### PR DESCRIPTION
**Pull request by Cein-Markey**
## what
Add new `iam_role_permissions_boundary` variable with a default of empty string and place it in `aws_iam_role.elasticsearch_user.permissions_boundary`

## why
This change is to allow users of the API to leverage/set permissions boundaries: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html

"A permissions boundary is an advanced feature for using a managed policy to set the maximum permissions that an identity-based policy can grant to an IAM entity"

### CODE REVIEW

**Comment by Cein-Markey** [comment]
Hey @nitrocode @adamcrews 👋  Just noticed there are some outstanding PRs that have been sat here a while and was wondering if there is an ETA on these. Thanks in advance and apologies for any disturbances.

File: variables.tf [hunk 1]
```diff
@@ -88,6 +88,12 @@ variable "iam_role_arns" {
   description = "List of IAM role ARNs to permit access to the Elasticsearch domain"
 }
 
+variable "iam_role_permissions_boundary" {
+  type        = string
+  default     = ""
```
[the remark this answers is not in the document]
Comment (Cein-Markey) [reply variables.tf:93]: Sure man 👍

[the remark this answers is not in the document]
Comment (Cein-Markey) [reply variables.tf]: @Gowiem Would you like me to change the other string defaults `""` ? I was just following the pattern of `variables.tf`.

[the remark this answers is not in the document]
Comment (Gowiem) [reply variables.tf]: @Cein-Markey you can keep the existing values -- Just moving forward we should use `null` to represent an empty value, so the chance you made is great.

**Review by Gowiem — APPROVED** [review]

File: variables.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cein-Markey) [reply variables.tf]: Ok cool man, thanks for the speedy review and approval 🤜

**Comment by Gowiem** [comment]
@Cein-Markey thanks for the contribution! Released as [0.34.0](https://github.com/cloudposse/terraform-aws-elasticsearch/releases/tag/0.34.0)!

**Pull request merged**